swap_meet/vendor.py

```diff
@@ -1,2 +1,67 @@
+from swap_meet import item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+    
+    def add(self, one_item):
+        if not one_item in self.inventory:
+            self.inventory.append(one_item)
+        return one_item
+    
+    def remove(self, one_item):
+        if not one_item in self.inventory:
+            return False
+        elif one_item in self.inventory:
+            self.inventory.remove(one_item)
+            return one_item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:   
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        for item in self.inventory:
+            if my_item in self.inventory and their_item in other_vendor.inventory:                
+                self.inventory.remove(my_item)
+                other_vendor.inventory.append(my_item)
+                other_vendor.inventory.remove(their_item)
+                self.inventory.append(their_item)
```

Comment:
In this situation, I would recommend using the instance methods that are on these instances of `Vendor` (`self` and `other_vendor`). Consider the following code:

```python
        if my_item in self.inventory and their_item in other_vendor.inventory:                
            self.remove(my_item)
            other_vendor.add(my_item)
            other_vendor.remove(their_item)
            self.add(their_item)
```

1. Even if it's subtle/simple, the instance methods `add` and `remove` have the logic and validation to add to the `inventory` specifically. When we do the line `other_vendor.inventory.append(...)`, it's adding more specific logic implementation to this function (`swap_item`), which is adding to the responsibility of the function `swap_item`... now it's responsible for swapping _and_ knowing how to add and remove items from the inventory (think about SRP/Single Responsibility Principle). In case we ever need to update/change how items are added and removed, we'd have to update `add`, `remove`, _and_ `swap_item`.
2. The code feels more readable in this way, with less `.`s to access things.